cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AccessTokenController.java

```diff
@@ -60,7 +60,7 @@ protected ModelAndView handleRequestInternal(final HttpServletRequest request, f
             // resource owner password grant type
             final J2EContext context = new J2EContext(request, response);
             final ProfileManager manager = new ProfileManager(context);
-            final OAuthUserProfile profile = (OAuthUserProfile) manager.get(true);
```

Comment:
I'd rather stay with the specific `OAuthUserProfile` than with a generic `UserProfile`. Because in that case, considering the authentication that must have occured, it must be an `OAuthUserProfile`.